ports/esp32/machine_i2c.c

```diff
@@ -146,12 +146,11 @@ static void machine_hw_i2c_print(const mp_print_t *print, mp_obj_t self_in, mp_p
 }
 
 mp_obj_t machine_hw_i2c_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
-    MP_MACHINE_I2C_CHECK_FOR_LEGACY_SOFTI2C_CONSTRUCTION(n_args, n_kw, all_args);
```

Comment:
I think we should still keep this legacy constructor behaviour.  And to get the new default-instance behaviour working as well, you can do:
```c
if (n_args != 0) {
    MP_MACHINE_I2C_CHECK_FOR_LEGACY_SOFTI2C_CONSTRUCTION(n_args, n_kw, all_args);
}
```
The cases are:
- `machine.I2C()` previously tried to create a SoftI2C instance but would fail because scl/sda are required arguments.  So this pattern won't be used in any existing working code.  With this PR it will now work and create the default hardware I2C instance.  I think that's an acceptable change that shouldn't break existing code.
- `machine.I2C(-1, ...)` will still work the same as before, allow creating SoftI2C instances.

Reply:
Great, thanks!